src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1174,13 +1144,13 @@ CHIP_ERROR CASESession::ConstructSaltSigmaR2(const ByteSpan & rand, const P256Pu
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR CASESession::ConstructSaltSigmaR3(const uint8_t * ipk, size_t ipkLen, MutableByteSpan & salt)
+CHIP_ERROR CASESession::ConstructSaltSigmaR3(const ByteSpan & ipk, MutableByteSpan & salt)
 {
     uint8_t md[kSHA256_Hash_Length];
     memset(salt.data(), 0, salt.size());
     Encoding::LittleEndian::BufferWriter bbuf(salt.data(), salt.size());
 
-    bbuf.Put(ipk, ipkLen);
+    bbuf.Put(ipk.data(), ipk.size());
```

Comment:
The hash should not be finished here either. I will update the SHA256 stream impl to enable "incremental digest".

Reply:
Updated.

Reply:
Removing the Begin() does not solve the problem. It's the usage of Finish() that is the problem, based on the overall transcript

Reply:
Updated to use GetDigest instead.